### ISSUE TITLE
How To Use dynamic Stateless

### ISSUE THREAD

**Issue by Mr-Mohammad** [dotnet-state-machine/stateless#554]
Hi,
What solution do you have for me to use stateless dynamically?

**Comment by Mr-Mohammad** [comment]
I want to design the process first with Json and then let Stateless execute it

**Comment by gius** [comment]
Hi, do you have any update on this topic? It would be great to be able to (de)serialize a state machine from JSON/XML.

I can imagine dynamically calling the fluent functions to build a state machine based on some template, but feels like a hack to me 🤷

**Comment by HenningNT** [comment]
Just to be clear, are we talking about storing some object state to database, and then reading in the object into the same state?
The state machine itself is stateless, so it supports de/serializing an object state, please see the JsonExample.

Are we talking about reading in a file that defines the state machine, the guards and actions?
I don't think Stateless will ever support that, unless someone decides they really want that feature, and builds it themselves.

**Comment by mclift** [comment]
I agree with @HenningNT that this isn't likely to be something Stateless would support as a built-in feature and may be better suited to a separate project. All of the state machines I've come across in my work have custom logic (e.g. `PermitIf`, `PermitDynamic`, etc.) with dependencies on the application's internals. It shouldn't be too difficult to write a parser to build very simple state machines from Mermaid, DOT graph or some other structured text format, but I don't see a path to building a parser that could include conditionals, dynamic transitions, entry behaviours, or any other custom logic without significant development effort.

All that said, there's a really interesting project idea there and I'd love to see what people would come up with.
